### PR TITLE
Improve continuous builder Git and macOS support

### PR DESCRIPTION
**Pull request by dberzano**
* Support `du` and `gdu` (on macOS with Homebrew)
* Fix repos update when pull caused an unwanted merge
* Fetch origin before resetting to an origin branch

### CODE REVIEW

File: ci/continuous-builder.sh [hunk 1]
```diff
@@ -114,6 +127,7 @@ while true; do
     git credential-store --file ~/.git-creds store
     git config --global credential.helper "store --file ~/.git-creds"
 
+    set-github-status -c ${STATUS_REF} -s $CHECK_NAME/pending || true
```
Comment (ktf) [inline ci/continuous-builder.sh]: Are you sure about this? We do not want to reset the status in case a PR is already green, no?

Comment (dberzano) [reply ci/continuous-builder.sh]: If I am not mistaken, `list-branch-pr` excludes PRs with a green state? I did not check it. In any case I have accidentally added this line to the PR, let me remove it for now.

Comment (dberzano) [reply ci/continuous-builder.sh]: I have removed the line in the new iteration.

**Comment by dberzano** [comment]
@brinick I have implemented your suggestion on `du/gdu`. This is the version currently in use on the macOS builder.

**Pull request merged**